src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -0,0 +1,424 @@
+    public void iCallTheSuspendVoucherAPIWithExpectedStatusOfAndStub(int responseCode, String stub) {
+        BaseStepDef.response = RestAssured.given()
+                .header("Content-Type", "application/json")
+                .queryParam("command", "suspend")
+                .header("registeringInstitutionId", registeringInstitutionId)
+                .header("X-CallbackURL", mockServer.getBaseUri()+stub)
+                .baseUri(voucherManagementConfig.voucherManagementContactPoint)
+                .body(suspendVoucherBody)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(responseCode).build())
+                .when()
+                .put(voucherManagementConfig.voucherLifecycleEndpoint)
+                .andReturn().asString();
+
+        redeemVoucherResponseBody = BaseStepDef.response;
+        logger.info("Identity Mapper Response: {}", BaseStepDef.response);
+    }
+
+    @And("I can create an VoucherRequestDTO for voucher reactivation")
+    public void iCanCreateAnVoucherRequestDTOForVoucherReactivation() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("{\n");
+        sb.append("    \"requestID\": \"849324499155\",\n");
+        sb.append("    \"batchID\": \"045155518258\",\n");
+        sb.append("    \"voucherInstructions\": [\n");
+        sb.append("        {\n");
+        sb.append("            \"serialNumber\": \"").append(serialNumber).append("\",\n");
+        sb.append("            \"status\": \"02\"\n");
+        sb.append("        }\n");
+        sb.append("    ]\n");
+        sb.append("}");
+
+        suspendVoucherBody = sb.toString();
+    }
+
+    @When("I call the validity check API with expected status of {int} and stub {string}")
+    public void iCallTheValidityCheckAPIWithExpectedStatusOfAndStub(int responseCode, String stub) {
+        BaseStepDef.response = RestAssured.given()
+                .header("Content-Type", "application/json")
+                .queryParam("serialNumber", serialNumber)
+                .queryParam("isValid", true)
+                .header("X-CallbackURL", mockServer.getBaseUri()+stub)
+                .baseUri(voucherManagementConfig.voucherManagementContactPoint)
+                .expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(responseCode).build())
+                .when()
+                .get(voucherManagementConfig.voucherValidityEndpoint)
+                .andReturn().asString();
+
+        redeemVoucherResponseBody = BaseStepDef.response;
+        logger.info("Identity Mapper Response: {}", BaseStepDef.response);
+    }
+
+    @And("I can extract result from validation callback and assert if validation is successful")
+    public void iCanExtractResultFromValidationCallbackAndAssertIfValidationIsSuccessful() {
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        //assertThat(allServeEvents).hasSize(1);
+        ServeEvent request = allServeEvents.get(0);
```

Comment:
can you write this as a util function and avoid using hardcoded indexes

Reply:
I am iterating through the list now to check for response